Extension/src/common.ts

```diff
@@ -534,6 +534,29 @@ export function checkInstallLockFile(): Promise<boolean> {
     return checkFileExists(getInstallLockPath());
 }
 
+/** Get the platform that the installed binaries belong to.*/
+export async function getInstalledBinaryPlatform(): Promise<string | undefined> {
+    // the debugAdapters folder is utilized to identify the platform
+    const path_to_check: string = getExtensionFilePath("debugAdapters");
+    let installedPlatform: string = "";
+    if (checkDirectoryExists(path_to_check)) {
+        const files: string[] = await readDir(path_to_check);
+        installedPlatform = "win32";
+        if (files !== undefined) {
+            (files).forEach((element: string) => {
```

Comment:
The only trouble with using `foreach` here is that there is no way to break out once you find what you are looking for.  Take a look a `some` which should allow you to return a `true` once you've set the `installedPlatform`.

Reply:
In the new approach, we are not reading the content of that folder anymore.